### DISCUSSION TITLE
Helmfile Support

### DISCUSSION THREAD

**Discussion by shibumi** [fluxcd/flux2#1087] [Proposals] [answered]
Hey everybody,
is there any plan to support helmfile in flux2 or does it already support helmfile? I know that flux has helm support, but I am not sure how this works with Helmfile. Can somebody update me about the current situation? :)

**Comment by stefanprodan** [comment, marked as answer]
We have no plans on supporting Helmfile as it overlaps with Flux current capabilities and is incompatible with how Flux is built (e.g. no shell-exec to Helm CLI, controller-runtime, etc).

**Reply by shibumi** [reply]
Ok this is good to know. Thanks a lot for your quick answer :) looks like flux is not really what I am looking for then.

**Reply by armenr** [reply]
+1 to this reply.

We'll have to settle for/stay with ArgoCD in that case. It's a real shame, since the developer/operator ergonomics around weaveworks gitops/fluxCD has been more enjoyable for our team than ArgoCD's has. 

@stefanprodan - In what way does it overlap? Is there feature equivalence/parity? If so, where can we find an article or info/documentation on that?

**Reply by hiddeco** [reply]
>  In what way does it overlap?

A Helmfile describes a declarative state, a `HelmRelease` (and `HelmRepository`, and `HelmChart`) also describe a declarative state. We maintain the latter built directly on top of Helm, and integrating this with the former is near impossible for reasons Stefan describes.

In more detail, I do not think the design of a Helmfile would be a great fit for Kubernetes in general (but mind, my experience with it is minimal and I am going by `README.md` details). With Kubernetes you tend to split up resources for composability and extensibility, and then "glue" them together using references. A Helmfile on the other hand seems to be all of this composed together in a single file (making e.g. reuse of things, etc. harder, or you would need to maintain shared state between files which seems not optimal nor great for multi-tenancy purposes).

**Reply by nullck** [reply]
@hiddeco  On helmfile we can make usage of "includes", like we can easily develop a base helmfile that can be included on different clusters helmfiles - https://github.com/roboll/helmfile#separating-helmfileyaml-into-multiple-independent-files
It's sad to learn that flux doesn't support helmfile.

**Reply by lodotek** [reply]
I agree. Nice thing about ArgoCD though, is that you can make it do whatever you want with it via plugins, ie https://github.com/travisghansen/argo-cd-helmfile
